exercicios/para-casa/resolucao.js

```diff
@@ -0,0 +1,92 @@
+
+
+function Cat(name, age, color, castrated, medicalHistory){
+    this.name = name,
+    this.age = age,
+    this.color = color,
+    this.castrated = castrated,
+    this.medicalHistory = medicalHistory
+}
+
+function Dog(name, age, color, castrated, race, medicalHistory)
+{
+    this.name = name,
+    this.age = age,
+    this.color = color,
+    this.castrated = castrated,
+    this.race = race,
+    this.medicalHistory = medicalHistory
+}
+
+function ExoticAnimal(name, age, color, species, adult, medicalHistory){
+    this.name = name,
+    this.age = age,
+    this.color = color,
+    this.species = species,
+    this.adult = adult,
+    this.medicalHistory = medicalHistory
+}
+
+function medicalHistory(){
+    this.medicalAppointment = []
+    this.vaccines = []
+    this.procedures = []
+    
+}
+
+
+
+Cat.prototype.miar = function miar(){
+    console.log(`O gato ${this.name} está miando!`)
+}
+
+Dog.prototype.latir = function latir(){
+    console.log(`O cachorro ${this.name} está latindo!`)
+}
+
+ExoticAnimal.prototype.ehAdult = function ehAdult(){
+    if (this.species === "hamster"){
```

Comment:
Eu teria colocado o "this.age <= 2" dentro da condição do if junto com o comparativo da espécie usando o &&, mas deve ser só jeitos diferentes de escrever código mesmo... Acredito que da forma como fez também esteja certa, comentei pq nunca tinha visto assim :)